HABApp/openhab/http_connection.py

```diff
@@ -320,6 +321,47 @@ async def async_get_item(self, item_name: str) -> dict:
         else:
             return await ret.json(encoding='utf-8')
 
+    async def async_remove_link(self, item_name, thing_UID, channel) -> bool:
+        if self.config.general.listen_only:
+            return False
+
+        # rest/links/ endpoint needs the channel to be url encoded (AAAA:BBBB:CCCC:0#NAME -> AAAA%3ABBBB%3ACCCC%3A0%23NAME)
+        channelUID = urllib.parse.quote(thing_UID + ":" + channel)
+        url = self.__get_openhab_url('rest/links/{item:s}/{channel:s}',item=item_name,channel=channelUID)
+        
+        fut = self.__session.delete(url,json={})
+        
+        ret = await self._check_http_response(fut)
+        return ret.status == 200
+
+    async def async_link_exists(self, item_name, thing_UID, channel) -> bool:
+        # rest/links/ endpoint needs the channel to be url encoded (AAAA:BBBB:CCCC:0#NAME -> AAAA%3ABBBB%3ACCCC%3A0%23NAME)
+        channelUID = urllib.parse.quote(thing_UID + ":" + channel)
+        url = self.__get_openhab_url('rest/links/{item:s}/{channel:s}',item=item_name,channel=channelUID)
+        
+        fut = self.__session.get(url,json={})
+
+        ret = await self._check_http_response(fut)
+        return ret.status == 200
+
+    async def async_add_link(self, item_name, thing_UID, channel) -> bool:
+        if self.config.general.listen_only:
+            return False
+
+        # check for item existence first, otherwhise OpenHAB creates a new item when we add a link with an unknown itemname
+        exists = await self.async_item_exists(item_name)        
+        if not exists:
+            return False
+        
+        # rest/links/ endpoint needs the channel to be url encoded (AAAA:BBBB:CCCC:0#NAME -> AAAA%3ABBBB%3ACCCC%3A0%23NAME)
+        channelUID = urllib.parse.quote(thing_UID + ":" + channel)
+        url = self.__get_openhab_url('rest/links/{item:s}/{channel:s}',item=item_name,channel=channelUID)
+        
+        fut = self.__session.put(url,json={})
```

Comment:
I also found this strange, but it gets the information from the url-path and if a payload exists, compares them with the payload and throws an error if they don't match.

See the [relevant code in the rest api](https://github.com/eclipse/smarthome/blob/e25f1bd548126ef44fbdc90dd1dd0f85e83bdef0/bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/link/ItemChannelLinkResource.java#L116).

After having a second look - the difference is you can't use the configuration parameter this way. I wasn't aware there are link-configurations at all, but it seems like there are different profiles one could use for setting up a link. The available profiles could be queried on the /profile-types endpoint. This seems to be an advanced feature though I personally never needed.

We could add an optional profile parameter (defaults to "system:default" and verify the given profile against the profile-types endpoint but this would not be enough if the configuration is extended in the future for example.

How do you want to support those, if at all?

Reply:
Can the payload be omitted? Then I'd go for that.

Reply:
It can be omitted yes, that is how it currently works.

But as said [here](https://github.com/spacemanspiff2007/HABApp/issues/150) we probably should allow passing the configuration when adding a link as Dict<str,Any>. 
I found posts online where people are forced to use the link configuration if e.g. specific bindings are involved. The configuration is very generic with Dict<str,Any> though. Not sure how this can be serialized/deserialized.
I think it would be the best move to make no assumptions of the configuration content at all, just pass it through and let the OpenHAB Api handle errors related to non-existing profiles or wrong configurations.

To make use of the new definition I'd also add a thing_get_link() to be able to access the configuration.